source/globals.d.ts

```diff
@@ -1,14 +1,61 @@
-declare const browser: AnyObject;
+type ArgumentTypes<TFunction> = TFunction extends (...args: infer TInferredArgs) => infer TInferredReturnType ? TInferredArgs : never;
 
-type AnyObject = {[key: string]: any};
+type FirstArg<TFunction extends any[]> =
+	TFunction extends [infer TFirstArg, ...any[]] ? TFirstArg :
+		TFunction extends [] ? undefined :
+			never;
+
+type TailArgs<TFunction extends any[]> = ((...args: TFunction) => any) extends ((
```

Comment:
Can you add some code comment docs for these types? They're not easily readable.

Are you sure they're even needed?

Reply:
Will do. It's used when v3 and v4 functions are created. It's to figure out what `...args` is. I believe @bfred-it mentioned there was a simpler solution in another PR, probably just jot using `...args`. I'll add a comment regardless.

Reply:
Awesome! Way more readable

https://github.com/sindresorhus/refined-github/blob/2a71a833754d53e649a67318aaa48776df2e053b/source/libs/api.ts#L30-L33